src/Features/CSharp/Portable/Completion/CompletionProviders/CrefCompletionProvider.cs

```diff
@@ -35,178 +43,267 @@ public override bool IsTriggerCharacter(SourceText text, int characterPosition,
             return CompletionUtilities.IsTriggerCharacter(text, characterPosition, options);
         }
 
-        protected override Task<bool> IsExclusiveAsync(Document document, int position, CompletionTriggerInfo triggerInfo, CancellationToken cancellationToken)
+        public override async Task ProduceCompletionListAsync(CompletionListContext context)
         {
-            return SpecializedTasks.True;
-        }
+            var document = context.Document;
+            var position = context.Position;
+            var options = context.Options;
+            var cancellationToken = context.CancellationToken;
 
-        protected override async Task<IEnumerable<CompletionItem>> GetItemsWorkerAsync(Document document, int position, CompletionTriggerInfo triggerInfo, System.Threading.CancellationToken cancellationToken)
-        {
             var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
             if (!tree.IsEntirelyWithinCrefSyntax(position, cancellationToken))
             {
-                return null;
+                return;
             }
 
-            var token = tree.FindTokenOnLeftOfPosition(position, cancellationToken);
-            token = token.GetPreviousTokenIfTouchingWord(position);
+            var token = tree.FindTokenOnLeftOfPosition(position, cancellationToken)
```

Comment:
Do we not have a helper for this pair of methods.  We have this pattern *everywhere*.   It should be something like GetPreviousCompleteToken